src/simulation.F90

```diff
@@ -349,13 +350,22 @@ subroutine finalize_batch()
 
     ! Write out state point if it's been specified for this batch
     if (statepoint_batch % contains(current_batch)) then
-      err = openmc_statepoint_write()
+      if (sourcepoint_batch % contains(current_batch) .and. source_write &
+           .and. .not. source_separate) then
+        err = openmc_statepoint_write(write_source=1)
+      else
+        err = openmc_statepoint_write(write_source=0)
+      end if
     end if
 
-    ! Write out source point if it's been specified for this batch
-    if ((sourcepoint_batch % contains(current_batch) .or. source_latest) .and. &
-         source_write) then
-      call write_source_point()
+    ! Write out a separate source point if it's been specified for this batch
+    if (sourcepoint_batch % contains(current_batch) .and. source_write &
+         .and. source_separate) call write_source_point()
+
+    ! Write a continously-overwritten source point if requested.
+    if (source_latest) then
+      filename = trim(path_output) // 'source' // '.h5'
+      call write_source_point(filename)
```

Comment:
The logic is indeed cleaner -- thanks for refactoring! Do we really write out a source point twice when `source_latest` is true?

And a thought: with the C API and the ability to access the source bank whenever, the `source_latest` feature doesn't seem very useful anymore.

Reply:
We really do!  Honestly, I question the utility of having a separate source file at all.  Are you open to the idea of removing the source.h5 files entirely?

Reply:
Sorry, forgot to respond to this before. Source files are still useful if you want to use them to start a new run. I would be open to removing them, but there needs to be a clear workflow for using a previous source for a new run.